libs/estdlib/src/gen_statem.erl

```diff
@@ -248,22 +248,22 @@ do_handle_state(EventType, Request, State) ->
             {noreply, State#state{
                 current_state = NextState,
                 data = NewData,
-                timer_map = maps:merge(TimerMap, TimerRefs)
+                timer_map = maps:merge(NewTimerMap, TimerRefs)
             }};
         {stop, Reason} ->
-            maybe_cancel_timer(CurrentState, TimerMap),
-            {stop, Reason, State};
+            NewTimerMap = maybe_cancel_timer(CurrentState, TimerMap),
```

Comment:
The rest of these are fine.  IIRC I left these not updating the `timer_map` because the gen_server would be going down, and the only consumer of the map is this module, but I can see how it is more correct to do it this way.